src/printer.js

```diff
@@ -2422,8 +2426,9 @@ function printNode(path, options, print) {
     }
     case "list":
     case "array": {
-      const open = node.shortForm ? "[" : concat([node.kind, "("]);
-      const close = node.shortForm ? "]" : ")";
+      const useShortForm = isSupported(options.phpVersion, "5.4");
```

Comment:
This will not work. Only arrays can use short form in 5.4, short form in `list()` assignments only work in 7.1.

Reply:
Oh right, good catch! Also, AST cleaning still needs to be adapted - I'll try to finish it tonight.

Reply:
I know, right? Skimming through the changed snapshots finally proved useful for once. 😅 

Trying to deepen my understanding here: What's AST cleaning?

Reply:
ast after prettier should be `===` ast before prettier

Reply:
Ah, right. Thanks for the explainer. 🙂

Reply:
So if we change `array()` to `[]` we need change ast before prettier, it is allow to catch problems with outputting broken ast

Reply:
Done. I reverted the behaviour for lists to how we used to do it - should we always use the short form for lists if phpVersion >= 7.1 though?

Reply:
Not sure. I have not used it often (neither the short nor the long form) and therefore still find the long form more intuitive since I've known it for way longer. But that's just me, and I think it would be worth considering the short form in the spirit of an optinionated formatter.

Maybe make a Twitter poll? 😄

Reply:
I'm also not using `list` much - let's leave it as is for now, I'll post the question in the issue about array short form, and maybe on twitter as well :wink: